yaml-schema.json

```diff
@@ -0,0 +1,138 @@
+{
+    "$schema": "http://json-schema.org/draft-07/schema",
+    "title": "Meta yaml",
+    "description": "Validate the meta yaml file for an nf-core module",
+    "type": "object",
+    "properties": {
+        "name": {
+            "type": "string",
+            "description": "Name of the module"
+        },
+        "description": {
+            "type": "string",
+            "description": "Description of the module"
+        },
+        "keywords": {
+            "type": "array",
+            "description": "Keywords for the module",
+            "items": {
+                "type": "string"
+            }
+        },
+        "authors": {
+            "type": "array",
+            "description": "Authors of the module",
+            "items": {
+                "type": "string"
+            }
+        },
+        "input": {
+            "type": "array",
+            "description": "Input channels for the module",
+            "items": {
+                "type": "object",
+                "patternProperties": {
+                    ".*": {
+                        "type": "object",
+                        "properties": {
+                            "type": {
+                                "type": "string",
+                                "description": "Type of the input channel"
```

Comment:
Should we set an enum based on: https://nf-co.re/docs/contributing/modules#documentation point 6?

Reply:
(applies to output: too)

Reply:
Added it now locally and I get quite a few deviations. Because I see quite a lot of "list" entries coming up, should we add it to the options? (if that is the correct groovy data object)

Reply:
Looking for an example, I found [justrma](https://github.com/nf-core/modules/blob/master/modules/nf-core/affy/justrma/main.nf) and the [yml](https://github.com/nf-core/modules/blob/master/modules/nf-core/affy/justrma/meta.yml). Looks like `celfiles_dir` should be `directory` instead of `list`, no?
I don't know if there are other modules with different situations.

Reply:
I would rather say `list` isn't so informative for the type, it's the contents of the list is more important (is it a list of files?)

Reply:
nested json schemas are not yet supported, but will be soon...

Reply:
I would opt for getting rid of `list` and just specifying the type of the list elements

Reply:
I think it is important to indicate if it is more then one item, so I would go for nested here, so we can actually check the items themselves as well.

Reply:
But shouldn't that be inthe description (that's normally how I do it)

type: file
description: A file or list of files that XYZ

Reply:
if we make it nested, we could then for example check the file extension of each individual file. More validation power 🦾

Reply:
:spaghetti: